examples/macosX64/src/macosX64Main/kotlin/main.kt

```diff
@@ -67,9 +100,43 @@ fun main() {
 					dirChosen = path ?: "none selected"
 					showDirPicker = false
 				}
-			}
 
+				SaveFilePicker(
+					show = showSaveFilePicker,
+					title = "Some title",
+					filename = "newTextFile",
+					fileExtension = "txt",
+				) { selectedFile ->
+					savedFile = selectedFile?.nsUrl?.path?.let { path ->
+						val result = writeToFile(path, "some nice text for our file")
+						result.getOrNull() == true
+					} ?: false
+					showSaveFilePicker = false
+				}
+			}
 		}
 	}
 	NSApp?.run()
 }
+
+@OptIn(ExperimentalForeignApi::class, BetaInteropApi::class)
+private fun writeToFile(filePath: String, contents: String): Result<Boolean> = runCatching {
+	val fileHandle = NSFileHandle.fileHandleForWritingAtPath(filePath)
```

Comment:
This does not work for me. `NSFileHandle.fileHandleForWritingAtPath(filePath)` returns null.

Reply:
To keep examples simpler, try https://github.com/Kotlin/kotlinx-io, it has common code, which can be used to save file on any platform. But anyway `NSFileHandle` is also okay.

Something like:
```kotlin
import io.ktor.utils.io.core.*
import kotlinx.io.buffered
import kotlinx.io.files.Path
import kotlinx.io.files.SystemFileSystem
import kotlinx.io.write
import kotlin.test.Test

class Learning {
    @OptIn(ExperimentalStdlibApi::class)
    @Test
    fun saveStuffToFile() {
        val file = Path("./test")

        SystemFileSystem.sink(file).buffered().use {
            it.write("1234".toByteArray())
        }
    }
}
```

Reply:
I have updated the example, but again I can't test it. Feel free to test it and merge it. If the path is returned "our job is done", although it would be great if the example worked properly.

Reply:
just tested and it works 🚀